agent/consul/catalog_endpoint_test.go

```diff
@@ -1322,6 +1336,20 @@ func TestCatalog_ListServices(t *testing.T) {
 	if out.Services["db"][0] != "primary" {
 		t.Fatalf("bad: %v", out)
 	}
+
+	t.Run("with option AllowNotModifiedResponse", func(t *testing.T) {
+		args.QueryOptions = structs.QueryOptions{
+			MinQueryIndex:            11,
+			MaxQueryTime:             20 * time.Millisecond,
+			AllowNotModifiedResponse: true,
+		}
+		err := msgpackrpc.CallWithCodec(codec, "Catalog.ListServices", &args, &out)
+		require.NoError(t, err)
+
+		require.Equal(t, out.Index, uint64(11))
```

Comment:
I would rather fetch the index once without AllowNotModifiedResponse, fetch the Index and re-do the request with AllowNotModified to avoid hardcording the index being 11 (or add first a requirement to be 11) since it would be hard to be sure it will always be 11 (think to changes in the future to anti-entropy or any change in Raft that increase this index to 12...)
We could then add another test with index -1 and check that out.NotModified is not set

Reply:
Oh ya, that probably explains why the test is failing in CI. I will see if I can change the test to start fewer things, so that the id is predictable. If that doesn't work then a lookup of the id makes sense.

Reply:
Refactoring to make the ID more predictable is going to be a much larger change, so I've used the Index from the previous query. Thank you for the suggestion!